jni/include/faiss_index_service.h

```diff
@@ -31,8 +31,50 @@ namespace faiss_wrapper {
 class IndexService {
 public:
     IndexService(std::unique_ptr<FaissMethods> faissMethods);
-    //TODO Remove dependency on JNIUtilInterface and JNIEnv
-    //TODO Reduce the number of parameters
+    /**
+     * Initialize index
+     *
+     * @param jniUtil jni util
+     * @param env jni environment
+     * @param metric space type for distance calculation
+     * @param indexDescription index description to be used by faiss index factory
+     * @param dim dimension of vectors
+     * @param numVectors number of vectors
+     * @param threadCount number of thread count to be used while adding data
+     * @param parameters parameters to be applied to faiss index
+     * @return pointer to the native index object
+     */
+    virtual jlong initIndex(knn_jni::JNIUtilInterface *jniUtil, JNIEnv *env, faiss::MetricType metric, std::string indexDescription, int dim, int numVectors, int threadCount, std::unordered_map<std::string, jobject> parameters);
```

Comment:
Ideally yes. Should I remove it or deprecate it?

Reply:
I think we should remove it, if it is not getting used. and if it is still getting used then marking it as deprecated will be right thing to do.

Reply:
I just tried this. There are a few integration tests that directly call this function. I don't think it would be too hard to change these to use the iterative method. Would this be best?

Reply:
Removed createIndex. Resolving.